--- a/addon-sdk/moz.build	2016-04-20 23:32:49.000000000 +0200
+++ b/addon-sdk/moz.build	2016-12-29 10:57:43.094144915 +0100
@@ -69,8 +69,8 @@
         'source/lib/sdk/tabs/helpers.js',
         'source/lib/sdk/tabs/namespace.js',
         'source/lib/sdk/tabs/observer.js',
-        'source/lib/sdk/tabs/tab-fennec.js',
         'source/lib/sdk/tabs/tab-firefox.js',
+        'source/lib/sdk/tabs/tab-fennec.js',
         'source/lib/sdk/tabs/tab.js',
         'source/lib/sdk/tabs/tabs-firefox.js',
         'source/lib/sdk/tabs/utils.js',
@@ -121,8 +121,8 @@
     ]
 
     EXTRA_JS_MODULES.commonjs.sdk.windows += [
-        'source/lib/sdk/windows/fennec.js',
         'source/lib/sdk/windows/firefox.js',
+        'source/lib/sdk/windows/fennec.js',
         'source/lib/sdk/windows/observer.js',
         'source/lib/sdk/windows/tabs-fennec.js',
     ]
