--- src/org/python/core/PySystemState.java
+++ src/org/python/core/PySystemState.java
@@ -667,9 +667,16 @@
         }
         try {
             String jythonpath = System.getenv("JYTHONPATH");
-            if (jythonpath != null) {
+            String pythonpath = System.getenv("PYTHONPATH");
+            if ((jythonpath != null) && (pythonpath == null)) {
                 registry.setProperty("python.path", jythonpath);
             }
+            else if ((jythonpath == null) && (pythonpath != null)) {
+                registry.setProperty("python.path", pythonpath);
+            }
+            else if ((jythonpath != null) && (pythonpath != null)) {
+                registry.setProperty("python.path", jythonpath + ":" + pythonpath);
+            }
         } catch (SecurityException e) {
         }
         registry.putAll(postProperties);
