--- a/configure.ac	2016-01-10 20:46:44.000000000 +0100
+++ b/configure.ac	2016-02-12 17:27:21.573923181 +0100
@@ -16,8 +16,8 @@
 
 PKG_CHECK_MODULES(GREETER, [
     liblightdm-gobject-1
-    gtk+-3.0
-    webkitgtk-3.0
+    gtk+-2.0
+    webkit-1.0
     dbus-glib-1
 ])
 
--- a/src/lightdm-webkit-greeter.c	2016-01-10 20:42:31.000000000 +0100
+++ b/src/lightdm-webkit-greeter.c	2016-02-12 17:31:07.781918565 +0100
@@ -224,14 +224,14 @@
 {
   gdouble opacity;
 
-  opacity = gtk_widget_get_opacity (window);
+  opacity = gtk_window_get_opacity (GTK_WINDOW (window));
   opacity -= 0.1;
   if (opacity <= 0)
     {
       gtk_main_quit ();
       return FALSE;
     }
-  gtk_widget_set_opacity (window, opacity);
+  gtk_window_set_opacity (GTK_WINDOW (window), opacity);
 
   return TRUE;
 }
