--- a/www-client/firefox/firefox-24.0-r1.ebuild	2013-10-12 22:06:13.218057063 +0200
+++ b/www-client/firefox/firefox-24.0-r1.ebuild	2013-10-12 22:37:42.203441496 +0200
@@ -44,7 +44,8 @@
 # More URIs appended below...
 SRC_URI="${SRC_URI}
 	http://dev.gentoo.org/~anarchy/mozilla/patchsets/${PATCH}.tar.xz
-	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/${PATCH}.tar.xz"
+	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/${PATCH}.tar.xz
+	https://launchpad.net/ubuntu/saucy/+source/${PN}/${PV}+build1-0ubuntu1/+files/${PN}_${PV}+build1-0ubuntu1.debian.tar.gz"
 
 ASM_DEPEND=">=dev-lang/yasm-1.1"
 
@@ -143,6 +144,7 @@
 }
 
 src_prepare() {
+	epatch "${WORKDIR}/debian/patches/unity-menubar.patch"
 	# Apply our patches
 	EPATCH_SUFFIX="patch"  	EPATCH_FORCE="yes" EOF
	git apply firefox-unity-support.patch
	rm firefox-unity-support.patch
	cat >> www-client/firefox/Manifest << EOF
DIST firefox_24.0+build1-0ubuntu1.debian.tar.gz 273652 SHA256 4e3f88622381def717b8f9d5c39aef916faadb88ced8c458b1496f187eadc390 SHA512 73dc738d2aab5f6dbc9faf60e61b0f1795db6ce773b13551207fc26a9ccd783a92cb0ca4f0c21f95e78b4ab74f9a62dc5d0497d2d92dca55f663d59e324b520d WHIRLPOOL 9f9aa0beb7fc69a97dd9381c88db5d62f9041dfed30f8470024f98a1a1d9cd0a8483e10ed018defd3c10673b84ddc55bdbe6d2e6049992264eeab818a36ad44e
