diff -U 3 -H -d -r -N -- a/make.globals b/make.globals
--- a/cnf/make.globals	2013-08-07 21:27:12.000000000 +0200
+++ b/cnf/make.globals	2013-09-26 18:00:58.175771100 +0200
@@ -19,7 +19,7 @@
 FFLAGS=""
 FCFLAGS=""
 
-SYNC="git://github.com/funtoo/ports-2012.git"
+SYNC="git://github.com/matijaskala/ports-2013.git"
 SYNC_USER="root"
 
 # Default distfiles mirrors. This rotation has multiple hosts and is reliable.
