diff -U 3 -H -d -r -N -- a/pym/portage/const.py b/pym/portage/const.py
--- a/pym/portage/const.py	2013-07-17 03:36:32.000000000 +0200
+++ b/pym/portage/const.py	2013-09-12 08:05:31.022081983 +0200
@@ -44,7 +44,7 @@
 
 # variables used with targetroot (these need to be absolute, but not
 # have a leading '/' since they are used directly with os.path.join on EROOT)
-VDB_PATH                 = "var/db/pkg"
+VDB_PATH                 = "usr/share/portage/pkg"
 CACHE_PATH               = "var/cache/edb"
 PRIVATE_PATH             = "var/lib/portage"
 WORLD_FILE               = PRIVATE_PATH + "/world"
diff -U 3 -H -d -r -N -- a/vartree.py b/vartree.py
--- a/pym/portage/dbapi/vartree.py	2013-09-21 08:52:38.000000000 +0200
+++ b/pym/portage/dbapi/vartree.py	2013-09-26 16:52:23.534211198 +0200
@@ -2280,7 +2280,7 @@
 
 				if not ignore:
 					if islink and f_match in \
-						("/lib", "/usr/lib", "/usr/local/lib"):
+						("/lib", "/usr/lib", "/usr/local/lib", "/bin", "/sbin", "/lib32", "/lib64"):
 						# Ignore libdir symlinks for bug #423127.
 						ignore = True
 
