--- a/gen_initramfs.sh
+++ b/gen_initramfs.sh
@@ -5,12 +5,12 @@
 
 _get_udevdir() {
     local udev_dir=$(pkg-config --variable udevdir udev)
-    [[ -n "${udev_dir}" ]] && echo $(realpath "${udev_dir}")
+    echo $(realpath "${udev_dir:-/lib/udev}")
 }
 
 _get_systemdutildir() {
     local systemdutil_dir=$(pkg-config systemd --variable=systemdutildir)
-    [[ -n "${systemdutil_dir}" ]] && echo $(realpath "${systemdutil_dir}")
+    echo $(realpath "${systemdutil_dir:-/lib/systemd}")
 }
 
 # The copy_binaries function is explicitly released under the CC0 license to
--- a/defaults/linuxrc
+++ b/defaults/linuxrc
@@ -257,6 +257,18 @@
         overlayfs)
             USE_OVERLAYFS=1
         ;;
+        multilive_subdir=*)
+            MULTILIVE_SUBDIR=${x#*=}
+        ;;
+        lang=*)
+            SLONTOO_LANG=${x#*=}
+        ;;
+        keytable=*)
+            SLONTOO_KEYMAP=${x#*=}
+        ;;
+        layout=*)
+            SLONTOO_XKB=${x#*=}
+        ;;
     esac
 done
 
--- a/defaults/initrd.d/00-livecd.sh
+++ b/defaults/initrd.d/00-livecd.sh
@@ -54,12 +54,15 @@
     local overlay=/mnt/overlay
     local static=/mnt/livecd
 
-    for i in "${overlay}" "${static}"; do
+    for i in "${overlay}" "${static}" "/mnt/multilivecd"; do
         [ ! -d "${i}" ] && mkdir -p "${i}"
     done
     good_msg "Loading aufs"
     modprobe aufs > /dev/null 2>&1
 
+    [ -n "${MULTILIVE_SUBDIR}" ] && \
+    mount -t squashfs -o loop,ro "${LOOP_PATH}" "/mnt/multilivecd" && \
+    mount -o rbind "/mnt/multilivecd/${MULTILIVE_SUBDIR}" "${static}" || \
     mount -t squashfs -o loop,ro "${LOOP_PATH}" "${static}"
     mount -t tmpfs none "${overlay}"
     mount -t aufs -o br:${overlay}:${static} aufs "${NEW_ROOT}"
@@ -81,12 +81,15 @@
     local workdir="${overlay}/.work"
     local static=/mnt/livecd
 
-    for i in "${overlay}" "${static}"; do
+    for i in "${overlay}" "${static}" "/mnt/multilivecd"; do
         [ ! -d "${i}" ] && mkdir -p "${i}"
     done
     good_msg "Loading overlayfs"
     modprobe overlay > /dev/null 2>&1
 
+    [ -n "${MULTILIVE_SUBDIR}" ] && \
+    mount -t squashfs -o loop,ro "${LOOP_PATH}" "/mnt/multilivecd" && \
+    mount -o rbind "/mnt/multilivecd/${MULTILIVE_SUBDIR}" "${static}" || \
     mount -t squashfs -o loop,ro "${LOOP_PATH}" "${static}"
     mount -t tmpfs none "${overlay}"
     mkdir "${upperdir}" "${workdir}"
@@ -235,17 +235,58 @@
         good_msg "Mounting squashfs & aufs filesystems"
         _setup_squashfs_aufs
         test_success "Mount filesystem"
+        if [ -n "${SLONTOO_LANG}" ] ; then
+            echo "LANG=${SLONTOO_LANG}.utf8" > "${NEW_ROOT}"/etc/env.d/02locale
+            sed -i /\\\<LANG\\\>.\'/d "${NEW_ROOT}"/etc/*.env
+            echo "export LANG='${SLONTOO_LANG}'" >> "${NEW_ROOT}"/etc/profile.env
+            echo "setenv LANG '${SLONTOO_LANG}'" >> "${NEW_ROOT}"/etc/csh.env
+            printf "[User]\nLanguage=${SLONTOO_LANG}.utf8\n" > "${NEW_ROOT}"/var/lib/AccountsService/users/$(grep :1000: "${NEW_ROOT}"/etc/passwd | cut -d: -f1)
+        fi
+        [ -n "${SLONTOO_KEYMAP}" ] && sed -i "s/keymap=.*/keymap=\"${SLONTOO_KEYMAP}\"/" "${NEW_ROOT}"/etc/conf.d/keymaps
+        if [ -n "${SLONTOO_XKB}" ] ; then
+            mkdir -p "${NEW_ROOT}"/etc/X11/xorg.conf.d
+            cat > "${NEW_ROOT}"/etc/X11/xorg.conf.d/00-keyboard.conf << EOF
+Section "InputClass"
+       Identifier "keyboard"
+       MatchIsKeyboard "yes"
+       Option "XkbLayout" "${SLONTOO_XKB}"
+EndSection
+EOF
+        fi
         return  # preserve old behaviour
     elif _is_overlayfs; then  # same for overlay fs.
         good_msg "Mounting squashfs & overlay fs filesystems"
         _setup_squashfs_overlayfs
         test_success "Mount filesystem"
+        if [ -n "${SLONTOO_LANG}" ] ; then
+            echo "LANG=${SLONTOO_LANG}.utf8" > "${NEW_ROOT}"/etc/env.d/02locale
+            sed -i /\\\<LANG\\\>.\'/d "${NEW_ROOT}"/etc/*.env
+            echo "export LANG='${SLONTOO_LANG}'" >> "${NEW_ROOT}"/etc/profile.env
+            echo "setenv LANG '${SLONTOO_LANG}'" >> "${NEW_ROOT}"/etc/csh.env
+            printf "[User]\nLanguage=${SLONTOO_LANG}.utf8\n" > "${NEW_ROOT}"/var/lib/AccountsService/users/$(grep :1000: "${NEW_ROOT}"/etc/passwd | cut -d: -f1)
+        fi
+        [ -n "${SLONTOO_KEYMAP}" ] && sed -i "s/keymap=.*/keymap=\"${SLONTOO_KEYMAP}\"/" "${NEW_ROOT}"/etc/conf.d/keymaps
+        if [ -n "${SLONTOO_XKB}" ] ; then
+            mkdir -p "${NEW_ROOT}"/etc/X11/xorg.conf.d
+            cat > "${NEW_ROOT}"/etc/X11/xorg.conf.d/00-keyboard.conf << EOF
+Section "InputClass"
+       Identifier "keyboard"
+       MatchIsKeyboard "yes"
+       Option "XkbLayout" "${SLONTOO_XKB}"
+EndSection
+EOF
+        fi
         return  # preserve old behaviour
     fi
 
     good_msg "Mounting squashfs filesystem"
 
     local squashfs_path="${LOOP_PATH}"
+    [ -n "${MULTILIVE_SUBDIR}" ] && \
+    mount -t squashfs -o loop,ro "${squashfs_path}" \
+        "/mnt/multilivecd" && \
+    mount -o rbind "/mnt/multilivecd/${MULTILIVE_SUBDIR}" \
+        "${NEW_ROOT}/mnt/livecd" || \
     mount -t squashfs -o loop,ro "${squashfs_path}" \
         "${NEW_ROOT}/mnt/livecd" || {
         bad_msg "squashfs filesystem could not be mounted."
