--- a/configure.ac
+++ b/configure.ac
@@ -299,7 +299,7 @@
 LIBS="$save_LIBS"
 AC_SUBST(CAP_LIBS)
 
-AC_CHECK_FUNCS([__secure_getenv secure_getenv])
+AC_CHECK_FUNCS([issetugid __secure_getenv secure_getenv])
 AC_CHECK_DECLS([
         memfd_create,
         gettid,
--- a/src/basic/missing.h
+++ b/src/basic/missing.h
@@ -528,7 +528,9 @@
 #ifndef HAVE_SECURE_GETENV
 #  ifdef HAVE___SECURE_GETENV
 #    define secure_getenv __secure_getenv
+#  elif defined HAVE_ISSETUGID
+#    define secure_getenv(...) (issetugid() ? NULL : getenv(__VA_ARGS__))
 #  else
 #    error "neither secure_getenv nor __secure_getenv are available"
 #  endif
 #endif
