=== modified file 'kde/window-decorator-kde4/kdecorationbridge.h'
--- kde/window-decorator-kde4/kdecorationbridge.h	2013-06-25 07:08:45 +0000
+++ kde/window-decorator-kde4/kdecorationbridge.h	2013-07-17 03:25:00 +0000
@@ -26,6 +26,7 @@
 #define KDECORATIONBRIDGE_H
 
 #include "kdecoration.h"
+#include "kdeversion.h"
 #include <QWidget>
 
 /**
@@ -42,7 +43,9 @@
     virtual bool isCloseable() const = 0;
     virtual bool isMaximizable() const = 0;
     virtual MaximizeMode maximizeMode() const = 0;
+#if KDE_IS_VERSION(4,10,90)
     virtual QuickTileMode quickTileMode() const = 0;
+#endif
     virtual bool isMinimizable() const = 0;
     virtual bool providesContextHelp() const = 0;
     virtual int desktop() const = 0;

=== modified file 'kde/window-decorator-kde4/window.cpp'
--- kde/window-decorator-kde4/window.cpp	2013-06-23 09:03:37 +0000
+++ kde/window-decorator-kde4/window.cpp	2013-07-17 03:25:00 +0000
@@ -226,11 +226,13 @@
     return wInfo.actionSupported (NET::ActionMax);
 }
 
+#if KDE_IS_VERSION(4,10,90)
 KDecoration::QuickTileMode
 KWD::Window::quickTileMode (void) const
 {
     return QuickTileNone;
 }
+#endif
 
 KDecoration::MaximizeMode
 KWD::Window::maximizeMode (void) const

=== modified file 'kde/window-decorator-kde4/window.h'
--- kde/window-decorator-kde4/window.h	2013-06-23 09:48:28 +0000
+++ kde/window-decorator-kde4/window.h	2013-07-17 03:25:00 +0000
@@ -71,7 +71,9 @@
 	virtual bool isCloseable (void) const;
 	virtual bool isMaximizable (void) const;
 	virtual MaximizeMode maximizeMode (void) const;
+#if KDE_IS_VERSION(4,10,90)
 	virtual QuickTileMode quickTileMode (void) const;
+#endif
 	virtual bool isMinimizable (void) const;
 	virtual bool providesContextHelp (void) const;
 	virtual int desktop (void) const;

