--- compiz/cmake/CompizGSettings.cmake	2012-12-11 16:35:00.479892153 +0100
+++ compiz/cmake/CompizGSettings.cmake	2012-12-11 16:35:00.492891989 +0100
@@ -52,7 +52,7 @@
 
 	# Recompile GSettings Schemas
 	install (CODE "
-		 execute_process (COMMAND cmake -DSCHEMADIR_USER=${_schemadir_user} -DSCHEMADIR_ROOT=${GSETTINGS_GLOBAL_INSTALL_DIR} -P ${COMPIZ_CMAKE_MODULE_PATH}/recompile_gsettings_schemas_in_dir_user_env.cmake)
+		 execute_process (COMMAND cmake -DSCHEMADIR_USER=${COMPIZ_DESTDIR}${_schemadir_user} -DSCHEMADIR_ROOT=${COMPIZ_DESTDIR}${GSETTINGS_GLOBAL_INSTALL_DIR} -P ${COMPIZ_CMAKE_MODULE_PATH}/recompile_gsettings_schemas_in_dir_user_env.cmake)
 		 ")
 
     endif (GSETTINGS_GLOBAL_INSTALL_DIR_SET)
