diff -U 3 -H -d -r -N -- compiz-1/compizconfig/ccsm/CMakeLists.txt compiz/compizconfig/ccsm/CMakeLists.txt
--- compiz-1/compizconfig/ccsm/CMakeLists.txt	2012-08-04 16:21:50.012726757 +0200
+++ compiz/compizconfig/ccsm/CMakeLists.txt	2012-08-04 16:22:14.612419219 +0200
@@ -17,7 +17,7 @@
 execute_process (COMMAND uname -p OUTPUT_VARIABLE PROC_ARCH)
 
 # Get Python Version
-execute_process (COMMAND python -c "import sys; print str (sys.version_info[0]) + '.' + str (sys.version_info[1])"
+execute_process (COMMAND python2 -c "import sys; print str (sys.version_info[0]) + '.' + str (sys.version_info[1])"
 		 OUTPUT_VARIABLE PY_VERSION)
 
 string (REPLACE "\n" "" PROC_ARCH ${PROC_ARCH})
@@ -37,7 +37,7 @@
 		  ${CMAKE_CURRENT_BINARY_DIR}/build/${PY_SCRIPTS_DIR}/ccsm)
 
 add_custom_command (OUTPUT ${PY_CCSM_LIB}
-		    COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build --version=${VERSION}
+		    COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build --version=${VERSION}
 		    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 		    COMMENT "Generating Python Bytecode")
 
@@ -46,7 +46,7 @@
 add_custom_target (ccsm_module ALL DEPENDS
 	     ${PY_CCSM_LIB})
 
-set (UNINSTALL_COMMAND "python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX} --version=${VERSION}")
+set (UNINSTALL_COMMAND "python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX} --version=${VERSION}")
 
 compiz_add_code_to_uninstall_target (${UNINSTALL_COMMAND} ${CMAKE_CURRENT_SOURCE_DIR})

diff -U 3 -H -d -r -N -- compiz-1/compizconfig/compizconfig-python/CMakeLists.txt compiz/compizconfig/compizconfig-python/CMakeLists.txt
--- compiz-1/compizconfig/compizconfig-python/CMakeLists.txt	2012-07-24 21:47:36.000000000 +0200
+++ compiz/compizconfig/compizconfig-python/CMakeLists.txt	2012-08-04 16:22:38.981114568 +0200
@@ -17,7 +17,7 @@
 execute_process (COMMAND uname -p OUTPUT_VARIABLE PROC_ARCH)
 
 # Get Python Version
-execute_process (COMMAND python -c "import sys; print str (sys.version_info[0]) + '.' + str (sys.version_info[1])"
+execute_process (COMMAND python2 -c "import sys; print str (sys.version_info[0]) + '.' + str (sys.version_info[1])"
 		 OUTPUT_VARIABLE PY_VERSION)
 
 string (REPLACE "\n" "" PROC_ARCH ${PROC_ARCH})
@@ -69,7 +69,7 @@
             ${PYTHON_LIBRARY_LOCAL}
 	     compizconfig)
 
-set (UNINSTALL_COMMAND "python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX} --version=${VERSION}")
+set (UNINSTALL_COMMAND "python2 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py uninstall --prefix=${CMAKE_INSTALL_PREFIX} --version=${VERSION}")
 
 compiz_add_code_to_uninstall_target (${UNINSTALL_COMMAND} ${CMAKE_CURRENT_BINARY_DIR})

