From 41092cc5de0384979fc211162e795719b3bfd114 Mon Sep 17 00:00:00 2001
From: infirit <infirit@gmail.com>
Date: Thu, 10 Oct 2013 14:44:10 +0200
Subject: [PATCH] POTFILES.in: Add missing files and make test work

---
 po/POTFILES.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/po/POTFILES.in b/po/POTFILES.in
index ab7f746..e12b340 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -64,6 +64,8 @@ plugins/externaltools/data/build.desktop.in
 plugins/externaltools/data/open-terminal-here.desktop.in
 plugins/externaltools/data/remove-trailing-spaces.desktop.in
 plugins/externaltools/data/run-command.desktop.in
+plugins/externaltools/data/search-recursive.desktop.in
+plugins/externaltools/data/switch-c.desktop.in
 plugins/filebrowser/filebrowser.pluma-plugin.desktop.in
 plugins/filebrowser/org.mate.pluma.plugins.filebrowser.gschema.xml.in.in
 plugins/filebrowser/pluma-file-bookmarks-store.c
-- 
1.8.4

