https://github.com/numpy/numpy/issues/3193

--- numpy/distutils/misc_util.py
+++ numpy/distutils/misc_util.py
@@ -605,11 +605,15 @@
     Linux, but not on OS X.
 
     """
-    so_ext = distutils.sysconfig.get_config_var('SO') or ''
-    # fix long extension for Python >=3.2, see PEP 3149.
-    if (not is_python_ext) and 'SOABI' in distutils.sysconfig.get_config_vars():
-        # Does nothing unless SOABI config var exists
-        so_ext = so_ext.replace('.' + distutils.sysconfig.get_config_var('SOABI'), '', 1)
+    if is_python_ext:
+        # EXT_SUFFIX in Python >=3.2.4 and >=3.3.1.
+        so_ext = distutils.sysconfig.get_config_var('EXT_SUFFIX') or distutils.sysconfig.get_config_var('SO') or ''
+    else:
+        # SHLIB_SUFFIX in Python >=3.2.4 and >=3.3.1.
+        so_ext = distutils.sysconfig.get_config_var('SHLIB_SUFFIX') or distutils.sysconfig.get_config_var('SO') or ''
+        # Fix extension for Python 3.2.0 - 3.2.3 and 3.3.0.
+        if distutils.sysconfig.get_config_var('SHLIB_SUFFIX') is None and 'SOABI' in distutils.sysconfig.get_config_vars():
+            so_ext = so_ext.replace('.' + distutils.sysconfig.get_config_var('SOABI'), '', 1)
 
     return so_ext
 
