--- numpy/distutils/system_info.py
+++ numpy/distutils/system_info.py
@@ -1369,19 +1369,12 @@
 
     def calc_info(self):
 
-        openblas_info = get_info('openblas')
-        if openblas_info:
-            self.set_info(**openblas_info)
-            return
-
         lapack_mkl_info = get_info('lapack_mkl')
         if lapack_mkl_info:
             self.set_info(**lapack_mkl_info)
             return
 
-        atlas_info = get_info('atlas_threads')
-        if not atlas_info:
-            atlas_info = get_info('atlas')
+        atlas_info = {}
 
         if sys.platform == 'darwin' and not atlas_info:
             # Use the system lapack from Accelerate or vecLib under OSX
@@ -1476,9 +1469,7 @@
             self.set_info(**openblas_info)
             return
 
-        atlas_info = get_info('atlas_blas_threads')
-        if not atlas_info:
-            atlas_info = get_info('atlas_blas')
+        atlas_info = {}
 
         if sys.platform == 'darwin'and not atlas_info:
             # Use the system BLAS from Accelerate or vecLib under OSX
