--- gnome-keyring-3.18.3/configure	2015-10-26 20:33:33.000000000 +0100
+++ gnome-keyring-3.18.3/configure	2016-09-26 01:16:55.388942264 +0200
@@ -675,8 +675,6 @@
 DBUS_SERVICES_DIR
 GCR_BASE_LIBS
 GCR_BASE_CFLAGS
-GCR_LIBS
-GCR_CFLAGS
 GCK_LIBS
 GCK_CFLAGS
 GSETTINGS_RULES
@@ -912,8 +910,6 @@
 GIO_LIBS
 GCK_CFLAGS
 GCK_LIBS
-GCR_CFLAGS
-GCR_LIBS
 GCR_BASE_CFLAGS
 GCR_BASE_LIBS
 P11_TESTS_CFLAGS
@@ -1619,8 +1615,6 @@
   GIO_LIBS    linker flags for GIO, overriding pkg-config
   GCK_CFLAGS  C compiler flags for GCK, overriding pkg-config
   GCK_LIBS    linker flags for GCK, overriding pkg-config
-  GCR_CFLAGS  C compiler flags for GCR, overriding pkg-config
-  GCR_LIBS    linker flags for GCR, overriding pkg-config
   GCR_BASE_CFLAGS
               C compiler flags for GCR_BASE, overriding pkg-config
   GCR_BASE_LIBS
@@ -15553,97 +15547,6 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
-fi
-
-pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for GCR" >&5
-$as_echo_n "checking for GCR... " >&6; }
-
-if test -n "$GCR_CFLAGS"; then
-    pkg_cv_GCR_CFLAGS="$GCR_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gcr-3 >= \$GCR_REQ\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gcr-3 >= $GCR_REQ") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_GCR_CFLAGS=`$PKG_CONFIG --cflags "gcr-3 >= $GCR_REQ" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$GCR_LIBS"; then
-    pkg_cv_GCR_LIBS="$GCR_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gcr-3 >= \$GCR_REQ\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gcr-3 >= $GCR_REQ") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_GCR_LIBS=`$PKG_CONFIG --libs "gcr-3 >= $GCR_REQ" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        GCR_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "gcr-3 >= $GCR_REQ" 2>&1`
-        else
-	        GCR_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "gcr-3 >= $GCR_REQ" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$GCR_PKG_ERRORS" >&5
-
-	as_fn_error $? "Package requirements (gcr-3 >= $GCR_REQ) were not met:
-
-$GCR_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables GCR_CFLAGS
-and GCR_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details." "$LINENO" 5
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	{ { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables GCR_CFLAGS
-and GCR_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details" "$LINENO" 5; }
-else
-	GCR_CFLAGS=$pkg_cv_GCR_CFLAGS
-	GCR_LIBS=$pkg_cv_GCR_LIBS
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-
 fi
 
 pkg_failed=no
--- gnome-keyring-3.18.3/configure.ac	2015-10-26 17:45:48.000000000 +0100
+++ gnome-keyring-3.18.3/configure.ac	2016-09-26 01:03:48.402958323 +0200
@@ -207,17 +207,14 @@
 # GCR and GCK libraries
 
 PKG_CHECK_MODULES(GCK, gck-1 >= $GCK_REQ)
-PKG_CHECK_MODULES(GCR, gcr-3 >= $GCR_REQ)
 PKG_CHECK_MODULES(GCR_BASE, gcr-base-3 >= $GCR_REQ)
 
 DAEMON_CFLAGS="$DAEMON_CFLAGS $GCK_CFLAGS $GCR_BASE_CFLAGS"
 DAEMON_LIBS="$DAEMON_LIBS $GCK_LIBS $GCR_BASE_LIBS"
 
 AC_SUBST(GCK_CFLAGS)
-AC_SUBST(GCR_CFLAGS)
 AC_SUBST(GCR_BASE_CFLAGS)
 AC_SUBST(GCK_LIBS)
-AC_SUBST(GCR_LIBS)
 AC_SUBST(GCR_BASE_LIBS)
 
 AC_DEFINE(GCK_API_SUBJECT_TO_CHANGE, 1, [Nod nod])
--- gnome-keyring-3.18.3/daemon/dbus/Makefile.am	2015-08-04 13:45:18.000000000 +0200
+++ gnome-keyring-3.18.3/daemon/dbus/Makefile.am	2016-09-26 01:12:21.028947863 +0200
@@ -109,7 +109,7 @@
 
 daemon_dbus_CFLAGS = \
 	$(DBUS_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 
 daemon_dbus_LIBS = \
 	libtestservice.la \
--- gnome-keyring-3.18.3/daemon/login/Makefile.am	2014-11-13 21:43:56.000000000 +0100
+++ gnome-keyring-3.18.3/daemon/login/Makefile.am	2016-09-26 01:11:55.317948387 +0200
@@ -10,7 +10,7 @@
 	$(NULL)
 libgkd_login_la_CFLAGS = \
 	$(GCK_CFLAGS) \
-	$(GCR_CFLAGS) \
+	$(GCR_BASE_CFLAGS) \
 	$(GOBJECT_CFLAGS)
 libgkd_login_la_LIBADD = \
 	$(GCK_LIBS) \
--- gnome-keyring-3.18.3/daemon/Makefile.am	2015-10-16 22:30:18.000000000 +0200
+++ gnome-keyring-3.18.3/daemon/Makefile.am	2016-09-26 01:11:40.394948692 +0200
@@ -64,7 +64,7 @@
 
 daemon_CFLAGS = \
 	$(GIO_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 
 daemon_LIBS = \
 	libgkd-control-client.la \
--- gnome-keyring-3.18.3/Makefile.in	2015-10-26 20:33:33.000000000 +0100
+++ gnome-keyring-3.18.3/Makefile.in	2016-09-26 01:13:05.782946949 +0200
@@ -1453,8 +1453,6 @@
 GCOV = @GCOV@
 GCR_BASE_CFLAGS = @GCR_BASE_CFLAGS@
 GCR_BASE_LIBS = @GCR_BASE_LIBS@
-GCR_CFLAGS = @GCR_CFLAGS@
-GCR_LIBS = @GCR_LIBS@
 GENHTML = @GENHTML@
 GETTEXT_PACKAGE = @GETTEXT_PACKAGE@
 GIO_CFLAGS = @GIO_CFLAGS@
@@ -1924,7 +1922,7 @@
 
 daemon_CFLAGS = \
 	$(GIO_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 
 daemon_LIBS = \
 	libgkd-control-client.la \
@@ -2033,7 +2031,7 @@
 
 daemon_dbus_CFLAGS = \
 	$(DBUS_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 
 daemon_dbus_LIBS = \
 	libtestservice.la \
@@ -2070,7 +2068,7 @@
 
 libgkd_login_la_CFLAGS = \
 	$(GCK_CFLAGS) \
-	$(GCR_CFLAGS) \
+	$(GCR_BASE_CFLAGS) \
 	$(GOBJECT_CFLAGS)
 
 libgkd_login_la_LIBADD = \
@@ -2717,16 +2715,14 @@
 	tool/gkr-tool-version.c
 
 gnome_keyring_@GKR_MAJOR@_CFLAGS = \
-	$(GTK_CFLAGS) \
 	$(GOBJECT_CFLAGS) \
 	$(GCK_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 
 gnome_keyring_@GKR_MAJOR@_LDADD = \
 	libegg.la \
 	$(GCK_LIBS) \
-	$(GCR_LIBS) \
-	$(GTK_LIBS) \
+	$(GCR_BASE_LIBS) \
 	$(GTHREAD_LIBS) \
 	$(GCRYPT_LIBS) \
 	$(DAEMON_LIBS)
--- gnome-keyring-3.18.3/tool/gkr-tool-import.c	2014-09-23 09:45:13.000000000 +0200
+++ gnome-keyring-3.18.3/tool/gkr-tool-import.c	2016-09-26 01:52:54.065898215 +0200
@@ -25,7 +25,7 @@
 #include "gkr-tool.h"
 
 #include <gck/gck.h>
-#include <gcr/gcr.h>
+#include <gcr/gcr-base.h>
 
 #include "egg/egg-hex.h"
 
--- gnome-keyring-3.18.3/tool/gkr-tool-trust.c	2014-09-23 09:45:13.000000000 +0200
+++ gnome-keyring-3.18.3/tool/gkr-tool-trust.c	2016-09-26 01:43:46.869909380 +0200
@@ -25,7 +25,7 @@
 #include "gkr-tool.h"
 
 #include <gck/gck.h>
-#include <gcr/gcr.h>
+#include <gcr/gcr-base.h>
 
 #include "egg/egg-hex.h"
 
--- gnome-keyring-3.18.3/tool/Makefile.am	2014-09-23 09:45:13.000000000 +0200
+++ gnome-keyring-3.18.3/tool/Makefile.am	2016-09-26 01:11:10.483949302 +0200
@@ -8,15 +8,13 @@
 	tool/gkr-tool-trust.c \
 	tool/gkr-tool-version.c
 gnome_keyring_@GKR_MAJOR@_CFLAGS = \
-	$(GTK_CFLAGS) \
 	$(GOBJECT_CFLAGS) \
 	$(GCK_CFLAGS) \
-	$(GCR_CFLAGS)
+	$(GCR_BASE_CFLAGS)
 gnome_keyring_@GKR_MAJOR@_LDADD = \
 	libegg.la \
 	$(GCK_LIBS) \
-	$(GCR_LIBS) \
-	$(GTK_LIBS) \
+	$(GCR_BASE_LIBS) \
 	$(GTHREAD_LIBS) \
 	$(GCRYPT_LIBS) \
 	$(DAEMON_LIBS)
