--- setup.py.in
+++ setup.py.in
@@ -6,8 +6,7 @@
 setup(name="slip", version="@VERSION@",
         py_modules=["slip.__init__", "slip.util.__init__",
             "slip.util.hookable", "slip.util.files",
-            "slip._wrappers.__init__", "slip._wrappers._gobject"],
-        requires=["selinux"])
+            "slip._wrappers.__init__", "slip._wrappers._gobject"])
 
 setup(name="slip.dbus", version="@VERSION@",
         py_modules=["slip.dbus.__init__", "slip.dbus.bus",
--- slip/util/files.py
+++ slip/util/files.py
@@ -24,7 +24,6 @@
            "overwrite_safely"]
 
 import os
-import selinux
 import shutil
 import tempfile
 import errno
@@ -134,9 +133,6 @@
 
     os.rename(dsttmpfile.name, dstpath)
 
-    if run_restorecon and selinux.is_selinux_enabled() > 0:
-        selinux.restorecon(dstpath)
-
 
 def linkorcopyfile(srcpath, dstpath, copy_mode_from_dst=True,
     run_restorecon=True):
@@ -234,9 +230,6 @@
 
     exists = os.path.exists(path)
 
-    if preserve_context and selinux.is_selinux_enabled() <= 0:
-        preserve_context = False
-
     try:
         fd, tmpname = tempfile.mkstemp(prefix=base + os.path.extsep,
                                        dir=dir_)
@@ -244,11 +237,6 @@
         if exists and preserve_mode:
             shutil.copymode(path, tmpname)
 
-        if exists and preserve_context:
-            ret, ctx = selinux.getfilecon(path)
-            if ret < 0:
-                raise RuntimeError("getfilecon(%r) failed" % path)
-
         f = os.fdopen(fd, "w")
         fd = None
 
@@ -259,12 +247,6 @@
 
         os.rename(tmpname, path)
 
-        if preserve_context:
-            if exists:
-                selinux.setfilecon(path, ctx)
-            else:
-                selinux.restorecon(path)
-
     finally:
         if f:
             f.close()
