--- wscript
+++ wscript
@@ -45,12 +45,20 @@
                 atleast_version=cairo_version_required,
                 args='--cflags --libs')
 
-# xpyb for Python 3 is not available yet.
-# the Python 3 version should probably have a different name than 'xpyb'
-#  ctx.check_cfg(package='xpyb',
-#                atleast_version=xpyb_version_required,
-#                args='--cflags --libs',
-#                mandatory=False)
+  if os.environ.get('PYCAIRO_DISABLE_XPYB') is None:
+    try:
+      p = subprocess.Popen(ctx.env['PYTHON'] + ['-c', 'import xcb'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+      p.wait()
+      p.stdout.close()
+      p.stderr.close()
+    except OSError:
+      pass
+    else:
+      if p.returncode == 0:
+        ctx.check_cfg(package='xpyb',
+                      atleast_version=xpyb_version_required,
+                      args='--cflags --libs',
+                      mandatory=False)
 
   # add gcc options
   if env['CC_NAME'] == 'gcc':
