https://github.com/mitsuhiko/babel/issues/45

--- babel/dates.py
+++ babel/dates.py
@@ -733,9 +733,9 @@
     In addition directional information can be provided that informs
     the user if the date is in the past or in the future:
 
-    >>> format_timedelta(timedelta(hours=1), add_direction=True)
+    >>> format_timedelta(timedelta(hours=1), add_direction=True, locale='en_US')
     u'In 1 hour'
-    >>> format_timedelta(timedelta(hours=-1), add_direction=True)
+    >>> format_timedelta(timedelta(hours=-1), add_direction=True, locale='en_US')
     u'1 hour ago'
 
     :param delta: a ``timedelta`` object representing the time difference to
--- babel/messages/mofile.py
+++ babel/messages/mofile.py
@@ -13,7 +13,7 @@
 import struct
 
 from babel.messages.catalog import Catalog, Message
-from babel._compat import range_type
+from babel._compat import PY2, range_type
 
 
 LE_MAGIC = 0x950412de
@@ -198,6 +198,10 @@
         koffsets += [l1, o1 + keystart]
         voffsets += [l2, o2 + valuestart]
     offsets = koffsets + voffsets
+    if PY2:
+        offsets_bytes = array.array("i", offsets).tostring()
+    else:
+        offsets_bytes = array.array("i", offsets).tobytes()
 
     fileobj.write(struct.pack('Iiiiiii',
         LE_MAGIC,                   # magic
@@ -206,4 +210,4 @@
         7 * 4,                      # start of key index
         7 * 4 + len(messages) * 8,  # start of value index
         0, 0                        # size and offset of hash table
-    ) + array.array("i", offsets).tostring() + ids + strs)
+    ) + offsets_bytes + ids + strs)
--- tests/test_numbers.py
+++ tests/test_numbers.py
@@ -173,7 +173,7 @@
 
 
 def test_get_currency_name():
-    assert numbers.get_currency_name('USD', 'en_US') == u'US dollars'
+    assert numbers.get_currency_name('USD', locale='en_US') == u'US Dollar'
 
 
 def test_get_currency_symbol():
