Index: nautilus-3.17.91/po/POTFILES.in
===================================================================
--- nautilus-3.17.91.orig/po/POTFILES.in
+++ nautilus-3.17.91/po/POTFILES.in
@@ -4,6 +4,8 @@
 data/org.gnome.Nautilus.appdata.xml.in
 data/nautilus-autorun-software.desktop.in.in
 data/org.gnome.Nautilus.desktop.in.in
+debian/mount-archive.desktop.in
+debian/nautilus-home.desktop.in
 eel/eel-canvas.c
 eel/eel-glib-extensions.c
 eel/eel-gnome-extensions.c
