---
 po/POTFILES.in |    2 ++
 1 file changed, 2 insertions(+)

--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -4,6 +4,8 @@ data/nautilus-autorun-software.desktop.i
 data/org.gnome.Nautilus.appdata.xml.in
 data/org.gnome.Nautilus.desktop.in
 data/org.gnome.nautilus.gschema.xml
+debian/mount-archive.desktop.in
+debian/nautilus-home.desktop.in
 eel/eel-canvas.c
 eel/eel-glib-extensions.c
 eel/eel-gnome-extensions.c
