Index: nautilus/po/POTFILES.in
===================================================================
--- nautilus.orig/po/POTFILES.in
+++ nautilus/po/POTFILES.in
@@ -5,6 +5,8 @@ data/org.gnome.Nautilus.appdata.xml.in
 data/nautilus-autorun-software.desktop.in.in
 data/org.gnome.Nautilus.desktop.in.in
 data/org.gnome.nautilus.gschema.xml
+debian/mount-archive.desktop.in
+debian/nautilus-home.desktop.in
 eel/eel-canvas.c
 eel/eel-glib-extensions.c
 eel/eel-gnome-extensions.c
