Index: nautilus-3.18.1/data/nautilus-autostart.desktop.in
===================================================================
--- nautilus-3.18.1.orig/data/nautilus-autostart.desktop.in
+++ nautilus-3.18.1/data/nautilus-autostart.desktop.in
@@ -4,3 +4,4 @@ Name=Files
 Exec=nautilus -n
 OnlyShowIn=GNOME;Unity;
 AutostartCondition=GSettings org.gnome.desktop.background show-desktop-icons
+NoDisplay=true
