Index: nautilus-3.18.1/data/org.gnome.Nautilus.desktop.in.in
===================================================================
--- nautilus-3.18.1.orig/data/org.gnome.Nautilus.desktop.in.in
+++ nautilus-3.18.1/data/org.gnome.Nautilus.desktop.in.in
@@ -16,6 +16,7 @@ X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
 X-GNOME-UsesNotifications=true
 Actions=Window;
+X-Unity-IconBackgroundColor=#af4853
 
 [Desktop Action Window]
 _Name=Open a New Window
