From b97f085565bac06d90a8577c3e3b9c2969ab3456 Mon Sep 17 00:00:00 2001
From: Sebastien Noël <sebastien@twolife.be>
Date: Tue, 28 Jan 2014 19:08:31 +0000
Subject: Restore backspace for going to parent folder

This restores the behaviour from nautilus <= 3.4.
---
Index: b/src/nautilus-window.c
===================================================================
--- a/src/nautilus-window.c
+++ b/src/nautilus-window.c
@@ -2012,6 +2012,7 @@
 	{ "back",  action_back },
 	{ "forward",  action_forward },
 	{ "up",  action_up },
+	{ "backspaceup",  action_up },
 	{ "action-menu", action_toggle_state_action_button, NULL, "false", NULL },
 	{ "reload", action_reload },
 	{ "stop", action_stop },
@@ -2095,6 +2096,9 @@
 	g_signal_connect_object (nautilus_file_undo_manager_get (), "undo-changed",
 				 G_CALLBACK (undo_manager_changed), window, G_CONNECT_SWAPPED);
 	undo_manager_changed (window);
+
+	/* Let backspace go up one dir as well */
+	nautilus_application_add_accelerator (app, "win.backspaceup", "BackSpace");
 }
 
 
