From 9ebf44120bf4c46217d50077acff7505d83fcfe9 Mon Sep 17 00:00:00 2001
From: infirit <infirit@gmail.com>
Date: Fri, 11 Oct 2013 11:50:39 +0200
Subject: [PATCH] Update POTFILES.in with changed file name in
 libmatepanelapplet-4.0 port

---
 po/POTFILES.in | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/po/POTFILES.in b/po/POTFILES.in
index 4b15b4a..94defaa 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -14,12 +14,9 @@ baobab/src/baobab-utils.c
 baobab/src/baobab-ringschart.c
 baobab/src/callbacks.c
 mate-dictionary/data/default.desktop.in
-mate-dictionary/data/MATE_DictionaryApplet.server.in.in
-mate-dictionary/data/MATE_DictionaryApplet.xml
 mate-dictionary/data/mate-dictionary.desktop.in.in
 mate-dictionary/data/org.mate.dictionary.gschema.xml.in
-[type: gettext/glade]mate-dictionary/data/mate-dictionary-preferences.ui
-[type: gettext/glade]mate-dictionary/data/mate-dictionary-source.ui
+mate-dictionary/data/org.mate.panel.applet.DictionaryAppletFactory.service.in
 mate-dictionary/data/spanish.desktop.in
 mate-dictionary/data/thai.desktop.in
 mate-dictionary/libgdict/gdict-client-context.c
@@ -43,6 +40,8 @@ mate-dictionary/src/gdict-sidebar.c
 mate-dictionary/src/gdict-source-dialog.c
 mate-dictionary/src/gdict-window.c
 mate-dictionary/src/main.c
+[type: gettext/glade]mate-dictionary/data/mate-dictionary-preferences.ui
+[type: gettext/glade]mate-dictionary/data/mate-dictionary-source.ui
 mate-screenshot/mate-screenshot.c
 mate-screenshot/mate-screenshot.desktop.in
 [type: gettext/glade]mate-screenshot/mate-screenshot.ui
-- 
1.8.3.2

