diff -U 3 -H -d -r -N -- a/portage-utils-0.30/main.c b/portage-utils-0.30/main.c
--- a/portage-utils-0.30/main.c	2013-04-30 06:08:48.000000000 +0200
+++ b/portage-utils-0.30/main.c	2013-09-12 09:09:36.623005356 +0200
@@ -692,7 +692,7 @@
 		_Q_EVS(STR,  PORTAGE_BINHOST,     binhost,             DEFAULT_PORTAGE_BINHOST)
 		_Q_EVS(STR,  PORTAGE_TMPDIR,      port_tmpdir,         CONFIG_EPREFIX "var/tmp/portage/")
 		_Q_EVS(STR,  PKGDIR,              pkgdir,              CONFIG_EPREFIX "usr/portage/packages/")
-		_Q_EVS(STR,  Q_VDB,               portvdb,             CONFIG_EPREFIX "var/db/pkg")
+		_Q_EVS(STR,  Q_VDB,               portvdb,             CONFIG_EPREFIX "usr/share/portage/pkg")
 		{ NULL, 0, _Q_BOOL, { NULL }, 0, NULL, }
 
 #undef _Q_EV
