--- src/zope/interface/_compat.py
+++ src/zope/interface/_compat.py
@@ -67,3 +67,12 @@
     def _dummy(*args):
         pass
     return _dummy
+
+try:
+    # Python 2 or >=3.2
+    callable = callable
+except NameError:
+    # Python 3.0 or 3.1
+    import collections
+    def callable(object):
+        return isinstance(object, collections.Callable)
--- src/zope/interface/declarations.py
+++ src/zope/interface/declarations.py
@@ -39,6 +39,7 @@
 from zope.interface.interface import Specification
 from zope.interface._compat import CLASS_TYPES as DescriptorAwareMetaClasses
 from zope.interface._compat import PYTHON3
+from zope.interface._compat import callable
 
 # Registry of class-implementation specifications
 BuiltinImplementationSpecifications = {}
--- src/zope/interface/verify.py
+++ src/zope/interface/verify.py
@@ -17,6 +17,7 @@
 from zope.interface.exceptions import BrokenMethodImplementation
 from types import FunctionType, MethodType
 from zope.interface.interface import fromMethod, fromFunction, Method
+from zope.interface._compat import callable
 import sys
 
 # This will be monkey-patched when running under Zope 2, so leave this
