From 1273b3f5d47385126a9dcbc192c17308cc867927 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Wed, 28 Sep 2016 14:32:47 +0200
Subject: [PATCH] Disable checks for new releases

---
 src/Helper/Settings/SettingRegistry.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Helper/Settings/SettingRegistry.cpp b/src/Helper/Settings/SettingRegistry.cpp
index 660bb3d..0fa4163 100644
--- a/src/Helper/Settings/SettingRegistry.cpp
+++ b/src/Helper/Settings/SettingRegistry.cpp
@@ -100,7 +100,7 @@ bool SettingRegistry::init(){
 	register_setting( Set::Player_Min2Tray, "min_to_tray", false );
 	register_setting( Set::Player_StartInTray, "start_in_tray", false );
 	register_setting( Set::Player_ShowTrayIcon, "show_tray_icon", true );
-	register_setting( Set::Player_NotifyNewVersion, "notify_new_version", true );
+	register_setting( Set::Player_NotifyNewVersion, "notify_new_version", false );
 	register_setting( Set::Player_SplitterState ,"splitter_state_player", QByteArray());
 	register_setting( Set::Player_Shortcuts, "shortcuts", RawShortcutMap());
 
-- 
2.10.0

