--- parole-0.8.0/src/gst/parole-gst.c	2015-02-28 15:23:54.000000000 +0100
+++ parole-0.8.0/src/gst/parole-gst.c	2016-09-29 13:31:33.708857023 +0200
@@ -2895,11 +2895,9 @@
             }
         }
 
-        ret = g_list_reverse (ret);
-
         if ( gst->priv->use_custom_subtitles == TRUE )
         {
-            ret = g_list_prepend (ret, g_strdup_printf("%s",gst->priv->custom_subtitles));
+            ret = g_list_append (ret, g_strdup_printf("%s",gst->priv->custom_subtitles));
         }
     }
     else
@@ -2908,7 +2906,7 @@
         return NULL;
     }
 
-    return ret;
+    return g_list_reverse (ret);
 }
 
 gboolean
