--- a/lib/libbamf/Makefile.in   2013-10-06 00:30:44.583163423 +0200
+++ b/lib/libbamf/Makefile.in   2013-10-06 00:31:44.861409841 +0200
@@ -465,9 +465,10 @@
 @HAVE_INTROSPECTION_TRUE@      $(NULL)

 @HAVE_INTROSPECTION_TRUE@Bamf_0_2_gir_INCLUDES = GObject-2.0 GLib-2.0
-@HAVE_INTROSPECTION_TRUE@Bamf_0_2_gir_CFLAGS = \
+@HAVE_INTROSPECTION_TRUE@Bamf_0_2_gir_SCANNERFLAGS = \
 @HAVE_INTROSPECTION_TRUE@      --warn-all \
-@HAVE_INTROSPECTION_TRUE@      --c-include='libbamf/libbamf.h' \
+@HAVE_INTROSPECTION_TRUE@      --c-include='libbamf/libbamf.h'
+@HAVE_INTROSPECTION_TRUE@Bamf_0_2_gir_CFLAGS = \
 @HAVE_INTROSPECTION_TRUE@      $(DBUS_CFLAGS) \
 @HAVE_INTROSPECTION_TRUE@      -I$(top_srcdir)/lib \
 @HAVE_INTROSPECTION_TRUE@      -I$(top_builddir)/lib \
