provide a way to disable libnss_systemd on musl
--- a/Makefile.am
+++ b/Makefile.am
@@ -5026,6 +5026,7 @@
 	test-nss
 
 # ------------------------------------------------------------------------------
+if HAVE_MYHOSTNAME
 libnss_systemd_la_SOURCES = \
 	src/nss-systemd/nss-systemd.sym \
 	src/nss-systemd/nss-systemd.c
@@ -5045,6 +5046,7 @@
 
 rootlib_LTLIBRARIES += \
 	libnss_systemd.la
+endif
 
 # ------------------------------------------------------------------------------
 if HAVE_MYHOSTNAME
