diff --git a/src/um-password-dialog.c b/src/um-password-dialog.c
index 5a763df..9b060e2 100644
--- a/src/um-password-dialog.c
+++ b/src/um-password-dialog.c
@@ -199,7 +199,7 @@ accept_password_dialog (GtkButton        *button,
                                 cursor = gdk_cursor_new_for_display (display, GDK_WATCH);
                                 gdk_window_set_cursor (gtk_widget_get_window (um->dialog), cursor);
                                 gdk_display_flush (display);
-                                g_object_unref (cursor);
+                                gdk_cursor_unref (cursor);
                                 return;
                         }
 
